content/tutorials/build-a-data-source-backend-plugin.md

```diff
@@ -0,0 +1,115 @@
+
+The folders and files used to build the backend for the data source are
+
+| file/folder         | description                                                                                                                                        |
+| ------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `Magefile.go`       | It’s not a requirement to use mage build files, but we strongly recommend using it so that you can use the build steps provided in the plugin SDK. |
+| `/src/plugins.json` | A JSON file describing the backend plugin                                                                                                          |
+| `/pkg/main.go`      | Starting point of the plugin binary.                                                                                                               |
+
+#### plugin.json
+
+When building a backend plugin these fields are important:
+
+| field      | description                                                                                                            |
+| ---------- | ---------------------------------------------------------------------------------------------------------------------- |
+| backend    | Should be set to `true` for backend plugins. This tells Grafana that it should start a binary when loading the plugin. |
+| executable | This is the name of the executable that Grafana expects to start.                                                      |
+| alerting   | Should be set to true if your backend datasource supports alerting.                                                    |
+
+In the next step we will look at the query endpoint!
+
+{{< /tutorials/step >}}
+{{< tutorials/step duration="2" title="Implement data queries" >}}
+
+We begin by opening the file `/pkg/datasource/sample-datasource.go`. In this file you will see the `DatasourceInstance` struct which implements the `backend.QueryDataHandler` interface.
+The `QueryData` method on this struct is where the data fetching happens for a data source plugin.
+
+Each request contains multiple queries to reduce traffic between Grafana and plugins.
+So you need to loop over the slice of queries, process each query, and then return the results of all queries.
+
+In the tutorial we have extracted a function named `query` to take care of each query model.
+Since each plugin has their own unique query model, Grafana sends it to the backend plugin as JSON. Therefore the plugin needs
+to `Unmarshal` the query model into something easier to work with.
+
+As you can see the sample only returns static numbers. Try to extend the plugin to return other types of data.
+You can read more about how to [build data frames in our docs](https://grafana.com/docs/grafana/latest/plugins/developing/dataframes).
+
+{{< /tutorials/step >}}
+{{< tutorials/step duration="1" title="Add Support for Health Checks" >}}
```

Comment:
[We should use sentence case for our titles](https://github.com/grafana/tutorials/pull/39/files#r424298675):
```suggestion
{{< tutorials/step duration="1" title="Add support for health checks" >}}
```